pkg/graveler/committed/merge_test.go

```diff
@@ -586,8 +587,216 @@ func Test_merge(t *testing.T) {
+				{action: actionTypeWriteRecord, key: "k7", identity: "dest:k7"},
+			},
+			expectedErr: nil,
+		},
+		"source and dest change same range conflict": {
+			baseRange: []testRange{
+				{rng: committed.Range{ID: "base:k1-k5", MinKey: committed.Key("k1"), MaxKey: committed.Key("k5"), Count: 5, EstimatedSize: 1234},
+					records: []testValueRecord{
+						{"k1", "base:k1"}, {"k2", "base:k2"}, {"k3", "base:k3"}, {"k4", "base:k4"},
+						{"k5", "base:k5"},
+					},
+				},
+			},
+			sourceRange: []testRange{
+				{rng: committed.Range{ID: "source:k3-k4", MinKey: committed.Key("k3"), MaxKey: committed.Key("k4"), Count: 2, EstimatedSize: 1234},
+					records: []testValueRecord{
+						{"k3", "base:k3"}, {"k4", "source:k4"},
+					},
+				},
+			},
+			destRange: []testRange{
+				{rng: committed.Range{ID: "dest:k6-k7", MinKey: committed.Key("k6"), MaxKey: committed.Key("k7"), Count: 2, EstimatedSize: 1234},
+					records: []testValueRecord{
+						{"k6", "dest:k6"}, {"k7", "dest:k7"},
+					},
+				},
+			},
+			conflictExpectedIdx: nil,
+			expectedActions:     []writeAction{},
+			expectedErr:         graveler.ErrConflictFound,
+		},
+		"no changes": {
+			baseRange: []testRange{
+				{rng: committed.Range{ID: "base:k3-k4", MinKey: committed.Key("k3"), MaxKey: committed.Key("k4"), Count: 2, EstimatedSize: 1234},
+					records: []testValueRecord{
+						{"k3", "base:k3"}, {"k4", "base:k4"},
+					},
+				},
+			},
+			sourceRange: []testRange{
+				{rng: committed.Range{ID: "base:k3-k4", MinKey: committed.Key("k3"), MaxKey: committed.Key("k4"), Count: 2, EstimatedSize: 1234},
+					records: []testValueRecord{
+						{"k3", "base:k3"}, {"k4", "base:k4"},
+					},
+				},
+			},
+			destRange: []testRange{
+				{rng: committed.Range{ID: "base:k3-k4", MinKey: committed.Key("k3"), MaxKey: committed.Key("k4"), Count: 2, EstimatedSize: 1234},
+					records: []testValueRecord{
+						{"k3", "base:k3"}, {"k4", "base:k4"},
+					},
+				},
+			},
+			conflictExpectedIdx: nil,
+			expectedActions: []writeAction{
+				{
+					action: actionTypeWriteRange,
+					rng:    committed.Range{ID: "base:k3-k4", MinKey: committed.Key("k3"), MaxKey: committed.Key("k4"), Count: 2, EstimatedSize: 1234},
+				},
+			},
+			expectedErr: nil,
+		},
+		"dest change key": {
+			baseRange: []testRange{
+				{
+					rng:     committed.Range{ID: "base:b-c", MinKey: committed.Key("b"), MaxKey: committed.Key("c"), Count: 2, EstimatedSize: 1024},
+					records: []testValueRecord{{key: "b", identity: "b"}, {key: "c", identity: "c"}},
+				},
+				{
+					rng:     committed.Range{ID: "base:d-e", MinKey: committed.Key("d"), MaxKey: committed.Key("e"), Count: 2, EstimatedSize: 1024},
+					records: []testValueRecord{{key: "d", identity: "d"}, {key: "e", identity: "e"}},
+				},
+			},
+			sourceRange: []testRange{
+				{
+					rng:     committed.Range{ID: "source:a-d", MinKey: committed.Key("a"), MaxKey: committed.Key("d"), Count: 2, EstimatedSize: 1024},
```

Comment:
I'm confused, the test is called dest change key?

Reply:
changed its name